packages/ai/core/types/ui-messages.ts

```diff
@@ -300,9 +306,21 @@ Keeps the last message when an error happens. Defaults to `true`.
   id?: string;
 
   /**
-   * Initial messages of the chat. Useful to load an existing chat history.
+   * Optional initialization object for the chat store.
    */
-  initialMessages?: Message[];
+  chats?: Record<string, Pick<ChatState, 'messages'>>;
+  /**
+   * Optional callback function that is called when chat store changes.
+   */
+  onChatStoreChange?: ({
+    event,
+    chatId,
+    state,
+  }: {
+    event: ChatStoreEvent;
+    chatId: string;
+    state: ChatState;
+  }) => void;
```

Comment:
do we want to expose this on useChat? what's the benefit? (for both `onChatStoreChange` and `chats`

Reply:
( I was thinking we pass in a `ChatStore` instead)